gcloud/monitoring/query.py

```diff
@@ -0,0 +1,661 @@
+
+    def __init__(self, metric_type):
+        self.metric_type = metric_type
+        self.group_id = None
+        self.projects = ()
+        self.resource_label_filter = None
+        self.metric_label_filter = None
+
+    def select_resources(self, *args, **kwargs):
+        """Select by resource labels.
+
+        See :meth:`Query.select_resources`.
+        """
+        self.resource_label_filter = _build_label_filter('resource',
+                                                         *args, **kwargs)
+
+    def select_metrics(self, *args, **kwargs):
+        """Select by metric labels.
+
+        See :meth:`Query.select_metrics`.
+        """
+        self.metric_label_filter = _build_label_filter('metric',
+                                                       *args, **kwargs)
+
+    def __str__(self):
+        filters = ['metric.type = "{type}"'.format(type=self.metric_type)]
+        if self.group_id is not None:
+            filters.append('group.id = "{id}"'.format(id=self.group_id))
+        if self.projects:
+            filters.append(
+                ' OR '.join('project = "{project}"'.format(project=project)
+                            for project in self.projects))
+        if self.resource_label_filter:
+            filters.append(self.resource_label_filter)
+        if self.metric_label_filter:
+            filters.append(self.metric_label_filter)
+
+        # Parentheses are never actually required, because OR binds more
+        # tightly than AND in the Monitoring API's filter syntax.
+        return ' AND '.join(filters)
+
+
+def _build_label_filter(category, *args, **kwargs):
+    """Construct a filter string to filter on metric or resource labels."""
+    terms = list(args)
+    for key in kwargs:
+        value = kwargs[key]
+        if value is None:
+            continue
+
+        suffix = None
+        if key.endswith('_prefix') or key.endswith('_suffix'):
+            key, suffix = key.rsplit('_', 1)
+
+        if category == 'resource' and key == 'resource_type':
+            key = 'resource.type'
+        else:
+            key = '.'.join((category, 'label', key))
+
+        if suffix == 'prefix':
+            term = '{key} = starts_with("{value}")'
+        elif suffix == 'suffix':
+            term = '{key} = ends_with("{value}")'
+        else:
+            term = '{key} = "{value}"'
+
+        terms.append(term.format(key=key, value=value))
+
+    return ' AND '.join(sorted(terms))
+
+
+def _format_timestamp(timestamp):
+    """Convert a datetime object to a string as required by the API.
+
+    :type timestamp: :class:`datetime.datetime`
+    :param timestamp: A datetime object.
+
+    :rtype: string
+    :returns: The formatted timestamp. For example:
+        ``"2016-02-17T19:18:01.763000Z"``
+    """
+    if timestamp.tzinfo is not None:
+        # Convert to UTC and remove the time zone info.
+        timestamp = timestamp.replace(tzinfo=None) - timestamp.utcoffset()
```

Comment:
You're suggesting replacing

    timestamp.isoformat() + 'Z'

with

    gcloud._helpers._datetime_to_rfc3339(timestamp)

I was kind of happy with the idea of dealing with timezones (just in case the user foolishly got involved with those) in conjunction with the formatting and not taking any kind of stand on them elsewhere in the code. The user can provide both the start time and the end time using `select_interval()`, by the way.